pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,176 @@
+    hash = "sha256-pH+zc7ck6rP3F3UQZHTtORYYvEH6HAbI2bJYcKaCFUU=";
+  };
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
-  ];
+  frontend = stdenv.mkDerivation {
+    pname = "dorion-ui";
+    inherit version src;
+
+    offlineCache = fetchYarnDeps {
+      yarnLock = "${src}/yarn.lock";
+      hash = "sha256-uvVD4ZJLGvXxiQSWSAoAcTv4tP7WKkgdOxCLYKhfhuM=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      prefetch-yarn-deps
+      yarn
+    ];
+
+    env.ESBUILD_BINARY_PATH = lib.getExe (esbuild.overrideAttrs (final: _: {
+      version = "0.15.18";
+      src = fetchFromGitHub {
+        owner = "evanw";
+        repo = "esbuild";
+        rev = "v${final.version}";
+        hash = "sha256-b9R1ML+pgRg9j2yrkQmBulPuLHYLUQvW+WTyR/Cq6zE=";
+      };
+      vendorHash = "sha256-+BfxCyg0KkDQpHt/wycy/8CTG6YBA/VJvJFhhzUnSiQ=";
+    }));
+
+    postPatch = ''
+      substituteInPlace package.json --replace 'pnpm prebuild' 'true'
+    '';
+
+    configurePhase = ''
+      runHook preConfigure
+
+      export HOME=$(mktemp -d)
+      yarn config --offline set yarn-offline-mirror $offlineCache
+      fixup-yarn-lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+
+      runHook postConfigure
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      cp ${shelter} src-tauri/injection/browser.js
+      yarn --offline build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      cp -r src-tauri/{injection,html} $out
+
+      runHook postInstall
+    '';
+  };
+
+in
+
+rustPlatform.buildRustPackage {
+  inherit version src pname;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "rsrpc-0.5.1" = "sha256-U5+86B/X5zQnlairXZ2mEaBlYx5q/w4Ko5kebv3ejDM=";
+      "tauri-plugin-window-state-0.1.0" = "sha256-AQt2cJ3ZP0Ffme2lfThcjSUA6FDE6srJryaNeMRXpz0=";
+    };
+  };
 
   nativeBuildInputs = [
+    pkg-config
     autoPatchelfHook
-    dpkg
     wrapGAppsHook
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk
     gst_all_1.gst-plugins-bad
     gst_all_1.gst-plugins-base
     gst_all_1.gst-plugins-good
-    webkitgtk
+    glib-networking
   ];
 
-  installPhase = ''
-    runHook preInstall
+  runtimeDependencies = [
+    libappindicator
+    libayatana-appindicator
+  ];
 
-    mkdir -pv $out
-    mv -v {bin,lib,share} $out
+  env.TAURI_RESOURCE_DIR = "${placeholder "out"}/lib";
 
-    runHook postInstall
+  postPatch = ''
+    substituteInPlace tauri.conf.json \
+      --replace '"beforeBuildCommand": "pnpm build",' '"beforeBuildCommand": "",' \
+      --replace '"distDir": "https://discord.com/app",' '"distDir": "src",' \
+      --replace "$(echo -e ',\n        "updater*"')" ""
```

Comment:
This substitution is not self-explanatory, and i think it would be far safer to put in a patch.

Reply:
Is this a blocker?

Reply:
At a very least a comment explaining what the `$(echo -e ',\n        "updater*"')` is supposed to fix, but i assume a patch would be more brief

Reply:
Done